src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -319,28 +337,41 @@
                                                                               image,
                                                                               creatableNetworkInterface);
         } else {
-            throw new RuntimeException("ERROR Operating System of type '" + operatingSystemType.toString() +
-                                       "' is not yet supported");
+            throw new RuntimeException(unsupportedOperatingSystemError(operatingSystemType.toString()));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)